src/components/Divider.tsx

```diff
@@ -1,70 +1,113 @@
-import React, { useEffect } from 'react'
-import { useDispatch } from 'react-redux'
-import { useSelector } from 'react-redux'
+import { useCallback, useLayoutEffect, useRef, useState } from 'react'
+import { useDispatch, useSelector } from 'react-redux'
 import { css } from '../../styled-system/css'
 import { SystemStyleObject } from '../../styled-system/types'
 import Path from '../@types/Path'
+import State from '../@types/State'
+import ThoughtId from '../@types/ThoughtId'
 import { setCursorActionCreator as setCursor } from '../actions/setCursor'
-import { DIVIDER_MIN_WIDTH, DIVIDER_PLUS_PX } from '../constants'
+import { DIVIDER_MIN_WIDTH } from '../constants'
 import attributeEquals from '../selectors/attributeEquals'
+import { getAllChildrenAsThoughts } from '../selectors/getChildren'
 import rootedParentOf from '../selectors/rootedParentOf'
+import editingValueUntrimmedStore from '../stores/editingValueUntrimmed'
+import viewportStore from '../stores/viewport'
 import fastClick from '../util/fastClick'
 import head from '../util/head'
+import isDivider from '../util/isDivider'
+
+/** Custom hook to fetch relevant thought IDs for calculating divider width. */
+const useRelevantThoughtIds = (path: Path): ThoughtId[] => {
+  return useSelector(
+    (state: State) => {
+      const parentPath = rootedParentOf(state, path)
+      const parentId = head(parentPath)
+      const grandParentPath = parentId ? rootedParentOf(state, parentPath) : []
```

Comment:
It won't affect the behavior here, but `[]` is not a valid `Path`. It should probably be set to `null` if there is no `parentId`.